pkgs/applications/networking/instant-messengers/discord/base.nix

```diff
@@ -1,93 +1,132 @@
-    ln -s $out/opt/${binaryName}/discord.png $out/share/pixmaps/${pname}.png
+      ln -s $out/opt/${binaryName}/${binaryName} $out/bin/
+      # Without || true the install would fail on case-insensitive filesystems
+      ln -s $out/opt/${binaryName}/${binaryName} $out/bin/${
+        lib.strings.toLower binaryName
+      } || true
+      ln -s $out/opt/${binaryName}/discord.png $out/share/pixmaps/${pname}.png
 
-    ln -s "${desktopItem}/share/applications" $out/share/
-  '';
+      ln -s "${desktopItem}/share/applications" $out/share/
+    '';
 
-  desktopItem = makeDesktopItem {
-    name = pname;
-    exec = binaryName;
-    icon = pname;
-    inherit desktopName;
-    genericName = meta.description;
-    categories = "Network;InstantMessaging;";
-    mimeType = "x-scheme-handler/discord";
+    desktopItem = makeDesktopItem {
+      name = pname;
+      exec = binaryName;
+      icon = pname;
+      inherit desktopName;
+      genericName = meta.description;
+      categories = "Network;InstantMessaging;";
+      mimeType = "x-scheme-handler/discord";
+    };
+
+    passthru.updateScript = writeScript "discord-update-script" ''
+      #!/usr/bin/env nix-shell
+      #!nix-shell -i bash -p curl gnugrep common-updater-scripts
+      set -eou pipefail;
+      url=$(curl -sI "https://discordapp.com/api/download/${
+        builtins.replaceStrings [ "discord-" "discord" ] [ "" "stable" ] pname
+      }?platform=linux&format=tar.gz" | grep -oP 'location: \K\S+')
+      version=''${url##https://dl*.discordapp.net/apps/linux/}
+      version=''${version%%/*.tar.gz}
+      update-source-version ${pname} "$version" --file=./pkgs/applications/networking/instant-messengers/discord/default.nix
+    '';
   };
+  darwin = stdenv.mkDerivation {
+    inherit pname version src meta;
+
+    nativeBuildInputs = [ undmg ];
 
-  passthru.updateScript = writeScript "discord-update-script" ''
-    #!/usr/bin/env nix-shell
-    #!nix-shell -i bash -p curl gnugrep common-updater-scripts
-    set -eou pipefail;
-    url=$(curl -sI "https://discordapp.com/api/download/${builtins.replaceStrings ["discord-" "discord"] ["" "stable"] pname}?platform=linux&format=tar.gz" | grep -oP 'location: \K\S+')
-    version=''${url##https://dl*.discordapp.net/apps/linux/}
-    version=''${version%%/*.tar.gz}
-    update-source-version ${pname} "$version" --file=./pkgs/applications/networking/instant-messengers/discord/default.nix
-  '';
+    sourceRoot = ".";
 
-  meta = with lib; {
-    description = "All-in-one cross-platform voice and text chat for gamers";
-    homepage = "https://discordapp.com/";
-    downloadPage = "https://discordapp.com/download";
-    license = licenses.unfree;
-    maintainers = with maintainers; [ ldesgoui MP2E ];
-    platforms = [ "x86_64-linux" ];
+    installPhase = ''
+      mkdir -p $out/Applications
+      cp -r *.app $out/Applications
+    '';
   };
-}
+in if stdenv.isLinux then linux else darwin
```

Comment:
I would rather move the "if linux, else darwin" logic into the default.nix and have this base.nix renamed `linux.nix`, and move the darwin builder into a `darwin.nix`

Reply:
fixed in 7f6bca5